mlflow/xgboost.py

```diff
@@ -173,13 +173,17 @@ def save_model(
     # Save an XGBoost model
     xgb_model.save_model(model_data_path)
 
-    conda_env, pip_requirements, pip_constraints = (
-        _process_pip_requirements(
-            get_default_pip_requirements(), pip_requirements, extra_pip_requirements,
+    if conda_env is None:
+        default_requirements = get_default_pip_requirements()
+        if not pip_requirements:
+            default_requirements = mlflow.infer_pip_requirements(
+                model_data_path, FLAVOR_NAME, fallback=default_requirements,
+            )
+        conda_env, pip_requirements, pip_constraints = _process_pip_requirements(
+            default_requirements, pip_requirements, extra_pip_requirements,
         )
-        if conda_env is None
-        else _process_conda_env(conda_env)
-    )
+    else:
+        conda_env, pip_requirements, pip_constraints = _process_conda_env(conda_env)
```

Comment:
Fixed the code to obtain `conda_env`, `pip_requirements`, and `pip_constraints` to avoid making the inference when `pip_requirements` is specified. I'll apply this change to other flavors.